packages/common/util.js

```diff
@@ -31,10 +31,13 @@ const log = require('./log');
  *
  * @alias module:util
  */
+const warned = new Set();
 exports.deprecate = (name, version, alternative) => {
-  let message = `${name} is deprecated after version ${version} and will be removed in a future release.`;
-  if (alternative) message += ` Use ${alternative} instead.`;
-  if (!('NO_DEPRECATION_WARNINGS' in process.env)) {
+  const key = `${name}-${version}`;
+  if (!warned.has(key)) {
```

Comment:
🐜 🐜 🐜 

By returning quickly, you don't need the following if-block.

```suggestion
  if (warned.has(key)) return;
```